odoo/addons/base/models/res_partner.py

```diff
@@ -243,6 +243,11 @@ def default_get(self, default_fields):
     commercial_company_name = fields.Char('Company Name Entity', compute='_compute_commercial_company_name',
                                           store=True)
     company_name = fields.Char('Company Name')
+
+    # show well placed warning when data is modified in "nearly related" fields
+    address_field_changed = fields.Char(compute="_compute_address_field_changed")
+    commercial_field_changed = fields.Char(compute="_compute_commercial_field_changed")
```

Comment:
since these are supposed to behave like related fields, they should always have the same value; if they don't, it means the user has changed the value in the current record (hence the warning). In light of that, i'm not sure why you feel the naming is bad (it does check if a field has been changed, since otherwise they must have similar values). But I could go with something else if you have another sensible candidate